### PR TITLE
Expand on the test-runs-query-builder component

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
See https://github.com/web-platform-tests/wpt.fyi/issues/535
(and, see #555 for the original component's definition.) 

This extends the builder's functionality to include adding + removing, and selecting a channel.

## Review Information
See http://localhost:8080/components/test/test-runs-query-builder.html for a playground.

### CODE REVIEW

File: webapp/components/product-info.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<dom-module id="browser-info">
```
Comment (lukebjerring) [inline webapp/components/product-info.html]: Browser info is just migrated code from `test-run` and `test-run-query` below, for re-use.

File: webapp/components/test-runs-query-builder.html [hunk 2]
```diff
@@ -7,38 +7,297 @@
...
+      </template>
+    </template>
     <div>
-      <template is="dom-repeat" items="[[products]]" as="product">
-        <display-logo product="[[product]]"></display-logo>
-        [[product.browser_name]]
-        <template is="dom-if" if="[[product.browser_version]]">
-          [[product.browser_version]]
-        </template>
+      <template is="dom-repeat" items="[[products]]" as="p" index-as="i">
```
Comment (lukebjerring) [inline webapp/components/test-runs-query-builder.html]: Note that nested fields on an object, which is inside an array, aren't observable from the parent, which is why they fields are separated in `product-builder`.

File: webapp/.eslintrc.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
   "globals": {
     "assert": true,
     "fixture": true,
+    "flush": true,
```
Comment (Hexcles) [inline webapp/.eslintrc.json]: Looks like we should also add `expect` here and remove `expect` from:
https://github.com/web-platform-tests/wpt.fyi/blob/06b80f2cf093ba41becf318bdd661890d8f82b88/webapp/components/test/test-runs-query.html#L13

**Comment by Hexcles** [comment]
Looks like I have to checkout this branch locally to play with the demo, since we don't deploy `test`.

File: webapp/.eslintrc.json (hunk 3 above)
Comment (lukebjerring) [reply webapp/.eslintrc.json]: Done.

**Comment by lukebjerring** [comment]
Yeah, unfortunately you'll need to run this locally for now, since the feature is not yet used.

File: webapp/components/test/test-runs-query-builder.html [hunk 4]
```diff
@@ -23,77 +23,79 @@
   </test-fixture>
 
   <script>
-    // Set up the playground.
-    document.querySelector('test-runs-query-builder').addProduct();
-    document.querySelector('test-runs-query-builder').addProduct();
+    document.addEventListener('WebComponentsReady', () => {
+      // Set up the playground.
+      document.querySelector('test-runs-query-builder').addProduct();
+      document.querySelector('test-runs-query-builder').addProduct();
```
Comment (Hexcles) [inline webapp/components/test/test-runs-query-builder.html]: I understand these setups need to happen after the component is ready, but do the tests also need to? We don't wait for components ready in other tests, so it seems a bit inconsistent.

File: webapp/components/test-runs-query.html [hunk 5]
```diff
@@ -12,17 +13,19 @@
      * Behaviour class for re-use of test-runs fetching behaviour.
      */
     // eslint-disable-next-line no-unused-vars
-    const TestRunsQuery = (superClass) => class extends QueryBuilder(superClass) {
+    const TestRunsQuery =
+      (superClass) => class extends QueryBuilder(ProductInfo(superClass)) {
       static get properties() {
         return {
           /* Parsed product objects, computed from the spec strings */
```
Comment (Hexcles) [inline webapp/components/test-runs-query.html]: Is this comment still true? IIUC `products` is now the parameter set by parents while `productSpecs` is the computed value.

Comment (lukebjerring) [reply webapp/components/test-runs-query.html]: Actually, both; if you set the product-specs value initially, it'll tee up the `products` array from those specs.

**Comment by lukebjerring** [comment]
Will address both of those comments in #580, which introduces the refresh button.

File: webapp/components/test/test-runs-query-builder.html (hunk 4 above)
Comment (lukebjerring) [reply webapp/components/test/test-runs-query-builder.html]: Eh, I just waited for it either way, but, reverted.

**Pull request merged**